src/quaternion_ops.jl

```diff
@@ -0,0 +1,187 @@
+#############################################################
+#
+#  add extra quaternion operations for Quaternions
+#  some should probably be merged back into Quaternions
+#
+#############################################################
+
+
+@doc """
+    Equivilent arbitrary axis rotation angle for a quaternion
+""" ->
+rot_angle(q::Quaternion) = 2 * acos(Quaternions.normalize(q).s)
+
+
+@doc """
+function to find the mean of a set of quaternians
+
+Args:
+
+    Qarray         - Vector of quaternions
+
+    method         - if 0 (default), the returned quaternion minimizes sum_i=1:n (||R(Qi) - R(Qbar)||)  with || the frobius norm, or
+                                     equivalently minimizes  sum_i=1:n (sin(phi_i / 2))^2, where phi_i is the angle of Qe_i, with  Qbar = Q_i x Qe_i
+
+                   - if 1, the returned quaternian minimizes sum_i=1:n (phi_i ^2) where phi_i is the angle of Qe_i, with  Qbar = Q_i x Qe_i
+                           this should be about the same as method == 0 if phi_i are all small
+""" ->
+function mean(Qarray::Vector{Quaternion}, method::Integer = 0)
+
+    if (method == 0)
+        M = zeros(4, 4)
+        for i = 1:length(Qarray)
+            Qi = convert(Vector{Float64}, Qarray[i])      # convert types to ensure we don't get quaternion multiplication
+            M[:,:] += Qi * (Qi')
+        end
+        evec = eigfact(Symmetric(M), 4:4)
+        Qbar = Quaternion(evec.vectors[1], evec.vectors[2], evec.vectors[3], evec.vectors[4], true)
+    else
+        error("I haven't coded this")
+    end
+
+    return Qbar
+
+end
+
+@doc """
+function to return the derivitive of the rotation matrix w.r.t the quaternion elements
+
+Args:
+
+    q            - a 4 element quaternion
+
+Returns:
+
+    dRdQ         - an array of four 3 x 3 matrices specifying each partial derivitive, i.e. dRdQ[i] = dR/d(Q[i])
+
+""" ->
+function quattorotderiv{T}(q::Quaternion{T})
```

Comment:
I was getting something different [here](https://github.com/FugroRoames/CoordinateTransformations.jl/blob/initial-code/src/commontransformations.jl#L165) in *CoordinateTransformations.jl*. (I have a unit test for this).

There *could* be a difference between assuming normalization or not. In fact, upon inspection, I think that is *exactly* what it is. Your derivative may be projected onto the tangent space of normalized quaternions - they will agree in this case but disagree when the quaternion becomes unnormalized. There's no right or wrong way to do this, but it should be documented so people can adjust their optimization strategies.